### PR TITLE
feat(google-rules-mvp): unified e2e tests v2 results

### PR DESCRIPTION
**Pull request by lisli1**
#### Details

Adding E2E tests for when the ATFA results feature flag is on, because the results_v2 are different.

##### Motivation

Adding/updating E2E tests to use the feature flag

##### Context

#4431 added the results_v2.json that is used for the added tests

#### Pull request checklist

- [x] Partially addresses an existing issue: WI 1853278
- [x] Ran `yarn fastpass`
- [n/a] Added/updated relevant unit test(s) (and ran `yarn test`)
- [n/a] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/electron/tests/automated-checks-view.test.ts [hunk 1]
```diff
@@ -18,9 +21,9 @@ describe('AutomatedChecksView', () => {
             'beforeEach',
         );
         app = await createApplication({ suppressFirstTimeDialog: true });
-        const resultsView = await app.openResultsView();
-        await resultsView.waitForScreenshotViewVisible();
-        cardsView = resultsView.createCardsViewController();
+        // resultsView = await app.openResultsView();
```
Comment (DaveTryon) [inline src/tests/electron/tests/automated-checks-view.test.ts:24]: Are these meant to be commented out?

Comment (lisli1) [reply src/tests/electron/tests/automated-checks-view.test.ts]: Yes, I was trying to test something out. I've uncommented them

File: src/tests/electron/tests/automated-checks-view.test.ts [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 import { scanForAccessibilityIssuesInAllModes } from 'tests/electron/common/scan-for-accessibility-issues';
 import { AppController } from 'tests/electron/common/view-controllers/app-controller';
 import { CardsViewController } from 'tests/electron/common/view-controllers/cards-view-controller';
+import { ResultsViewController } from 'tests/electron/common/view-controllers/results-view-controller';
 import { commonAdbConfigs, setupMockAdb } from 'tests/miscellaneous/mock-adb/setup-mock-adb';
 
 describe('AutomatedChecksView', () => {
     let app: AppController;
     let cardsView: CardsViewController;
+    let resultsView: ResultsViewController;
```
Comment (DaveTryon) [inline src/tests/electron/tests/automated-checks-view.test.ts:15]: Can this be scoped inside `openResultsAndCardsViews`?

Comment (lisli1) [reply src/tests/electron/tests/automated-checks-view.test.ts:15]: Oh, right. Good catch! Updating

File: src/tests/electron/tests/results-view.test.ts [hunk 3]
```diff
@@ -178,7 +230,13 @@ describe('ResultsView', () => {
     });
 
     it('export report button exists', async () => {
+        await openResultsView();
         await setupWindowForCommandBarReflowTest('wide');
         await resultsView.waitForSelector(ResultsViewSelectors.exportReportButton);
     });
+
+    async function openResultsView(): Promise<void> {
```
Comment (DaveTryon) [inline src/tests/electron/tests/results-view.test.ts:238]: Potential refactor (maybe a separate PR): If this returned `Promise<ResultsViewController>`, then resultsView could be defined inside each test case.

File: src/tests/electron/tests/automated-checks-view.test.ts [hunk 4]
```diff
@@ -67,7 +72,44 @@ describe('AutomatedChecksView', () => {
...
+        expect(await cardsView.queryRuleGroupContents()).toHaveLength(1);
+        await cardsView.assertExpandedRuleGroup(3, 'ImageViewName', 1);
+    });
+
+    it('should pass accessibility validation in all contrast modes', async () => {
+        await openResultsAndCardsViews();
         await scanForAccessibilityIssuesInAllModes(app);
     });
+
+    async function openResultsAndCardsViews(): Promise<void> {
```
Comment (DaveTryon) [inline src/tests/electron/tests/automated-checks-view.test.ts:109]: Potential refactor (maybe a separate PR): If this returned `Promise<ResultsViewController>`, then `resultsView` could be scoped inside each test method.

**Review by DaveTryon — APPROVED** [review]
Added some observations about scoping

**Pull request merged**